tools/src/test/java/org/apache/kafka/tools/consumer/group/ConsumerGroupServiceTest.java

```diff
@@ -0,0 +1,299 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class ConsumerGroupServiceTest {
+    public static final String GROUP = "testGroup";
+
+    public static final int NUM_PARTITIONS = 10;
+
+    private static final List<String> TOPICS = IntStream.range(0, 5).mapToObj(i -> "testTopic" + i).collect(Collectors.toList());
+
+    private static final List<TopicPartition> TOPIC_PARTITIONS = TOPICS.stream()
+            .flatMap(topic -> IntStream.range(0, NUM_PARTITIONS).mapToObj(i -> new TopicPartition(topic, i)))
+            .collect(Collectors.toList());
+
+    private final Admin admin = mock(Admin.class);
+
+    @Test
+    public void testAdminRequestsForDescribeOffsets() {
+        String[] args = new String[]{"--bootstrap-server", "localhost:9092", "--group", GROUP, "--describe", "--offsets"};
+        ConsumerGroupCommand.ConsumerGroupService groupService = consumerGroupService(args);
+
+        when(admin.describeConsumerGroups(ArgumentMatchers.eq(Collections.singletonList(GROUP)), any()))
+                .thenReturn(describeGroupsResult(ConsumerGroupState.STABLE));
+        when(admin.listConsumerGroupOffsets(ArgumentMatchers.eq(listConsumerGroupOffsetsSpec()), any()))
+                .thenReturn(listGroupOffsetsResult(GROUP));
+        when(admin.listOffsets(offsetsArgMatcher(), any()))
+                .thenReturn(listOffsetsResult());
+
+        Tuple2<Option<String>, Option<Seq<ConsumerGroupCommand.PartitionAssignmentState>>> res = groupService.collectGroupOffsets(GROUP);
+        assertEquals(Some.apply("Stable"), res._1);
+        assertTrue(res._2.isDefined());
+        assertEquals(TOPIC_PARTITIONS.size(), res._2.get().size());
+
+        verify(admin, times(1)).describeConsumerGroups(ArgumentMatchers.eq(Collections.singletonList(GROUP)), any());
+        verify(admin, times(1)).listConsumerGroupOffsets(ArgumentMatchers.eq(listConsumerGroupOffsetsSpec()), any());
+        verify(admin, times(1)).listOffsets(offsetsArgMatcher(), any());
+    }
+
+    @Test
+    public void testAdminRequestsForDescribeNegativeOffsets() {
+        String[] args = new String[]{"--bootstrap-server", "localhost:9092", "--group", GROUP, "--describe", "--offsets"};
+        ConsumerGroupCommand.ConsumerGroupService groupService = consumerGroupService(args);
+
+        TopicPartition testTopicPartition0 = new TopicPartition("testTopic1", 0);
+        TopicPartition testTopicPartition1 = new TopicPartition("testTopic1", 1);
+        TopicPartition testTopicPartition2 = new TopicPartition("testTopic1", 2);
+        TopicPartition testTopicPartition3 = new TopicPartition("testTopic2", 0);
+        TopicPartition testTopicPartition4 = new TopicPartition("testTopic2", 1);
+        TopicPartition testTopicPartition5 = new TopicPartition("testTopic2", 2);
+
+        // Some topic's partitions gets valid OffsetAndMetadata values, other gets nulls values (negative integers) and others aren't defined
```

Comment:
Is this comment correct? Or the test? I don't see the negative values

Reply:
Renamed

Reply:
Actually, I just copied it from the scala version of test.
Removed mention about negative values.

Reply:
The test is called testAdminRequestsForDescribeNegativeOffsets so we might want to figure out if that's the original intention.

Reply:
Looked in git log and this comment here from the moment test added - so I think we want to keep it as is, because we just move them to java - https://github.com/apache/kafka/commit/d95c1919458bd5621774394c9eb61698ce2187b8#diff-fcf6c81f0d8445b938d80a9d1f5ef838f731807d27b3f22a2cfefe2762b3001eR78

Reply:
Oh I see -- at this point the negative value is already converted to null. The comment was confusing.

See https://github.com/apache/kafka/pull/8057/files or https://github.com/apache/kafka/blob/7e5ef9b509a00973d271def5f70ca2062208e778/clients/src/main/java/org/apache/kafka/clients/admin/internals/ListConsumerGroupOffsetsHandler.java#L144